src/rgw/rgw_rest_s3.cc

```diff
@@ -1858,10 +1858,11 @@ int RGWPutObj_ObjStore_S3::get_params()
   append = s->info.args.exists("append");
   if (append) {
     string pos_str = s->info.args.get("position");
-    if (pos_str.empty()) {
+    string err;
+    position = (uint64_t)strict_strtoll(pos_str.c_str(), 10, &err);
```

Comment:
this would allow for negative values, would convert them (which is probably worse than setting them to zero).
can you add `strict_strtoull()` ?

Reply:
Sorry for late reply. Strict_strtoull will also convert negative values. I modify the code so that negative values are not allowed here.